Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIDeclarativeNetRequestCocoa.mm

```diff
@@ -138,9 +141,31 @@
     }, extensionContext().identifier());
 }
 
-void WebExtensionAPIDeclarativeNetRequest::getDynamicRules(Ref<WebExtensionCallbackHandler>&& callback)
+void WebExtensionAPIDeclarativeNetRequest::getDynamicRules(NSDictionary *filter, Ref<WebExtensionCallbackHandler>&& callback)
 {
-    WebProcess::singleton().sendWithAsyncReply(Messages::WebExtensionContext::DeclarativeNetRequestGetDynamicRules(), [protectedThis = Ref { *this }, callback = WTFMove(callback)](Expected<String, WebExtensionError>&& result) {
+    NSString *outExceptionString;
+
+    static NSDictionary<NSString *, id> *keyTypes = @{
+        getDynamicRulesRuleIDsKey: @[ NSNumber.class ]
+    };
+
+    if (!validateDictionary(filter, nil, nil, keyTypes, &outExceptionString)) {
+        callback->reportError(outExceptionString);
+        return;
+    }
+
+    NSMutableSet *otherFilterKeys = [NSMutableSet setWithArray:filter.allKeys];
+    [otherFilterKeys minusSet:[NSSet setWithObject:getDynamicRulesRuleIDsKey]];
+    if (otherFilterKeys.count) {
+        callback->reportError(toErrorString(@"declarativeNetRequest.getDynamicRules()", nullString(), [NSString stringWithFormat:@"unexpected property: '%@' in 'filter' parameter", otherFilterKeys.allObjects.firstObject]));
```

Comment:
```suggestion
        callback->reportError(toErrorString(@"declarativeNetRequest.getDynamicRules()", "filter"_s, [NSString stringWithFormat:@"unexpected property '%@'", otherFilterKeys.allObjects.firstObject]));
```

Reply:
We don't usually error on unexpected properties in Safari, so I would just remove this altogether. I'm pushing all other browsers to do similar in the WECG.